tests/unit-test.el

```diff
@@ -141,31 +141,132 @@
+              :to-equal "js")))
 
   (it "returns js for js alias in cljc file."
-    (expect (cljr--with-clojure-temp-file "foo.cljc"
-              (insert "(ns foo)")
-              (cljr--unresolved-alias-ref "js"))
-            :to-equal "js")))
+    (cljr--with-clojure-temp-file "foo.cljc"
+      (insert "(ns foo)")
+      (expect (cljr--unresolved-alias-ref "js")
+              :to-equal "js"))))
+
+(defmacro with-point-at (text &rest body)
+  (declare (indent 1))
+  `(progn (insert ,text)
+          (goto-char (point-min))
+          (re-search-forward "|")
+          (delete-char -1)
+          ,@body))
+
+(describe "cljr--language-context-at-point"
+  (it "identifies a clj file"
+    (cljr--with-clojure-temp-file "foo.clj"
+      (insert "(ns foo)")
+      (expect (cljr--language-context-at-point)
+              :to-equal '("clj" nil))))
+
+  (it "identifies a cljs file"
+    (cljr--with-clojure-temp-file "foo.cljs"
+      (insert "(ns foo)")
+      (expect (cljr--language-context-at-point)
+              :to-equal '("cljs" nil))))
+
+  (it "identifies a cljc file"
+    (cljr--with-clojure-temp-file "foo.cljc"
+      (insert "(ns foo)")
+      (expect (cljr--language-context-at-point)
+              :to-equal '("cljc" nil))))
+
+  (it "identifies a cljc file with a cljs context"
+    (cljr--with-clojure-temp-file "foo.cljc"
+      (with-point-at "(ns foo) #?(:cljs (Math/log|))"
+        (expect (cljr--language-context-at-point)
+                :to-equal '("cljc" "cljs")))))
+
+  (it "identifies a cljc file with a clj context"
+    (cljr--with-clojure-temp-file "foo.cljc"
+      (with-point-at "(ns foo) #?(:clj (Math/log|))"
+        (expect (cljr--language-context-at-point)
+                :to-equal '("cljc" "clj"))))))
+
+(describe "cljr--prompt-or-select-libspec"
+  (it "prompts user for namespace selection"
+    (spy-on 'completing-read :and-return-value "[a.a :as a]")
+    (expect (cljr--prompt-or-select-libspec '("[a.a :as a]"
+                                              "[a.b :as b]"))
+            :to-equal "[a.a :as a]"))
+
+  (it "returns user content from prompt regardless of candidates"
+    (spy-on 'completing-read :and-return-value "[alpha :as a]")
+    (expect (cljr--prompt-or-select-libspec
+             '("[a.a :as a]"
+               "[a.b :as b]"))
+            :to-equal "[alpha :as a]"))
+
+  (it "short-circuits if only one candidate matches"
+    (expect 'completing-read :to-have-been-called-times 0)
+    (expect (cljr--prompt-or-select-libspec '("[a.a :as a]"))
+            :to-equal "[a.a :as a]"))
+
+  (it "prompts anyway if `cljr-magic-requires' is `:prompt'"
+    (spy-on 'completing-read :and-return-value "[a.a :as a]")
+    (let ((cljr-magic-requires :prompt))
+      (expect (cljr--prompt-or-select-libspec '("[a.a :as a]"))
+              :to-equal "[a.a :as a]"))))
+
+(describe "cljr-slash"
+  (before-each (setq cljr-slash-uses-suggest-libspec t))
+  (after-each (setq cljr-slash-uses-suggest-libspec nil))
```

Comment:
Let's set it to `<whatever the previous value was>` instead of `nil`

Reply:
This is behavior is broken in buttercup anyway. It's supposed to support using :var ((cljr-slash-uses-suggest-libspec t)) as a lexical let around the entire describe but it just doesn't appear to work at present. I just swapped it with a local lexical let for each cljr-slash invocation to enable the feature for each call.

Reply:
Could you paste something similar as a `;; comment`?

Linking to a buttercup issue or such

Reply:
I don't have a buttercup issue for it, it just didn't work. I just changed it to use:

```
(let ((cljr-slash-uses-suggest-libspec t))
          (cljr-slash))
```

For those three test cases. That's correct though a little more verbose. If someone wants to clean it up or understands buttercup better so they can get it to work they are welcome to, I don't have the time to tweak it though, sorry.